subworkflows/local/cutadapt_workflow.nf

```diff
@@ -53,8 +53,25 @@ workflow CUTADAPT_WORKFLOW {
         CUTADAPT_SUMMARY_MERGE ( "copy", CUTADAPT_SUMMARY.out.tsv )
     }
 
+    //Filter empty files
+    ch_trimmed_reads
+        .branch {
+            failed: it[0].single_end ? it[1].size() < 1.KB : it[1][0].size() < 1.KB
+            passed: it[0].single_end ? it[1].size() >= 1.KB : it[1][0].size() >= 1.KB
+        }
+        .set { ch_trimmed_reads_result }
+    ch_trimmed_reads_result.passed.set { ch_trimmed_reads_passed }
+    ch_trimmed_reads_result.failed
+        .map { meta, reads -> [ meta.id ] }
+        .collect()
+        .subscribe {
+            samples = it.join(",")
+            log.error "Samples $samples had too low read count after trimming with cutadapt. Please check whether the correct primer sequences for trimming were supplied. If you want to ignore that issue and continue running the pipeline, use `--ignore_failed_samples`."
+            params.ignore_failed_samples ? { log.warn "Ignoring failed samples and continue!" } : System.exit(1)
+        }
+
     emit:
-    reads   = ch_trimmed_reads
```

Comment:
This puzzles me, you changed the name of the output, but where do you have the corresponding read from the channel with the new name?

Reply:
```
    ch_trimmed_reads
        .branch {
```
this means that the channel will be branched/splitted
```
            failed: it[0].single_end ? it[1].size() < 1.KB : it[1][0].size() < 1.KB
            passed: it[0].single_end ? it[1].size() >= 1.KB : it[1][0].size() >= 1.KB
        }
        .set { ch_trimmed_reads_result }
```
and here it specifies the names of the branches (`failed` and `passed`) and the conditions what samples goes where and puts it in a new channel `ch_trimmed_reads_result`.
`  ch_trimmed_reads_result.passed.set { ch_trimmed_reads_passed }`
takes than all elements in the` ch_trimmed_reads_result.passed` channel into `ch_trimmed_reads_passed`, which is the end result. The other lines are just the error message.

Was that the question?